### PR TITLE
routerrpc: Add detailed info logging during a rescan

### PR DESCRIPTION
**Pull request by kornpow**
## Change Description
https://github.com/lightningnetwork/lnd/issues/3658
During a rescan, which can happen for a variety of reasons, there is a section where LND is querying the chain backend for blocks and filtering transactions in order to update its DB. Prior to this change, the entire time that it spends querying the backend, there are little to no useful logs being generated, even at lower debuglevels.

This change tells the user via the log file what block height it is currently scanning so they get the sanity check that their node is not broken and stuck in a loop.

Things like block hash could also be logged, but at a lower level (debug/trace) if people desire that.

## Steps to Test
Steps for reviewers to follow to test the change.
Trigger a wallet rescan:
Start LND with this flag:
`lnd --reset-wallet-transactions`

### Testing
- [ ] Your PR passes all CI checks.
- [ ] Tests covering the positive and negative (error paths) are included.
- [ ] Bug fixes contain tests triggering the bug to prevent regressions.

### Code Style and Documentation
- [ ] The change obeys the [Code Documentation and Commenting](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#CodeDocumentation) guidelines, and lines wrap at 80.
- [ ] Commits follow the [Ideal Git Commit Structure](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#IdealGitCommitStructure).
- [X] Any new logging statements use an appropriate subsystem and logging level.
- [ ] Any new lncli commands have appropriate tags in the comments for the rpc in the proto file.
- [ ]  [There is a change description in the release notes](https://github.com/lightningnetwork/lnd/tree/master/docs/release-notes), or `[skip ci]` in the commit message for small changes.

📝 Please see our  [Contribution Guidelines](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md) for further guidance.

### CODE REVIEW

File: routing/router.go
Comment (guggero) [inline routing/router.go:857]: Should we give a bit more info on what exactly this is doing? Otherwise during normal operations, this is a bit useless. So perhaps something like "Filtering blocks for closed channels, next height: %v"?

**Review by guggero — COMMENTED** [review]
Thanks for the improvement!

File: routing/router.go
Comment (kornpow) [reply routing/router.go]: I updated this, with a similar message to your suggestion

**Review by guggero — APPROVED** [review]
Nice, LGTM :tada:

**Comment by kornpow** [comment]
Should I run the go formatter? to fix the linter?
I ran `gofmt [-w] routing/router.go` and nothing happened

**Comment by guggero** [comment]
Ah, it's the line length. You need to manually break before the argument to not exceed 80 characters (while counting the tab character as 8 chars). Unfortunately the formatter can't do that automatically.

**Comment by kornpow** [comment]
Ok, fixed manually

**Comment by kornpow** [comment]
Besides the failing cross-compilation I need one more review plz @ProofOfKeags or @positiveblue?

**Comment by kornpow** [comment]
One thing to note is the `--reset-wallet-transactions` flag doesnt excercise this code like I thought.

The rescan operation which triggered me to starting working on this PR was "triggered" by me switching from ZMQ bitcoind to RPC polling Bitcoind.

**Review by positiveblue — APPROVED** [review]
LGTM, ever if the log is not added in he flow that you had in mind, having this logs at debug/trace level is a good thing.


You can add more for the flow that you had in this PR or leave it for another one 👍

**Comment by Roasbeef** [comment]
Reran CI here, it failed as our build cache was full.

**Review by Roasbeef — APPROVED** [review]
LGTM 👠

**Comment by kornpow** [comment]
My name is in the release notes, so I think so! 🎉

**Pull request merged**
